driver/src/driver/stablex_driver.rs

```diff
@@ -0,0 +1,171 @@
+use crate::contracts::stablex_contract::StableXContract;
+use crate::error::DriverError;
+use crate::price_finding::PriceFinding;
+
+use std::collections::HashSet;
+
+use web3::types::U256;
+
+pub struct StableXDriver<'a, C: StableXContract> {
```

Comment:
No you are right, we can use `dyn StableXContract`. I only did it like this because the other components did it (which I believe was because in the beginning we didn't know better).

Good call.